ui/apps/dashboard/src/components/Metrics/Dashboard.tsx

```diff
@@ -0,0 +1,143 @@
+'use client';
+
+import { useState } from 'react';
+import { RangePicker } from '@inngest/components/DatePicker';
+import type { RangeChangeProps } from '@inngest/components/DatePicker/RangePicker.jsx';
+import EntityFilter from '@inngest/components/Filter/EntityFilter';
+import { Pill } from '@inngest/components/Pill';
+import {
+  useBatchedSearchParams,
+  useBooleanSearchParam,
+  useSearchParam,
+  useStringArraySearchParam,
+} from '@inngest/components/hooks/useSearchParam';
+import {
+  durationToString,
+  parseDuration,
+  subtractDuration,
+  toDate,
+  type DurationType,
+} from '@inngest/components/utils/date';
+import { RiArrowDownSFill, RiArrowRightSFill } from '@remixicon/react';
+import { useQuery } from 'urql';
+
+import { GetBillingPlanDocument } from '@/gql/graphql';
+import { FailedFunctions } from './FailedFunctions';
+import { FunctionStatus } from './FunctionStatus';
+
+type EntityType = {
+  id: string;
+  name: string;
+};
+
+export const DEFAULT_DURATION = { hours: 24 };
+
+const getFrom = (start?: Date, duration?: DurationType | '') =>
+  start || subtractDuration(new Date(), duration ? duration : DEFAULT_DURATION);
+
+const getDefaultRange = (start?: Date, end?: Date, duration?: DurationType | '') =>
+  start && end
+    ? {
+        type: 'absolute' as const,
+        start: start,
+        end: end,
+      }
+    : {
+        type: 'relative' as const,
+        duration: duration ? duration : DEFAULT_DURATION,
+      };
+
+export const Dashboard = ({
+  apps = [],
+  functions = [],
+}: {
+  apps?: EntityType[];
+  functions?: EntityType[];
+}) => {
+  const [selectedApps, setApps, removeApps] = useStringArraySearchParam('apps');
```

Comment:
I've serialized all filter settings to url params so that any given chart can be shared or reloaded. If the serialization makes the UI sluggish I can add an interim step to state, but it seems ok for now.